tests/integration/test_l1_read_request.py

```diff
@@ -0,0 +1,283 @@
+import re
+import unittest
+from callee import Regex
+from common.constants import MESIState, LogLevel
+from utils.event_handler import handle_event
+from tests.integration.integration_setup import IntegrationSetup
+
+
+class TestCommandL1ReadRequestData(IntegrationSetup):
+
+    def setUp(self):
+        super().setUp()
+        # Default to 0 for trace event op code.
+        # This allows us to create a child class for the instruction read requests,
+        # which should have identical behavior.
+        self.trace_event = 0
+        # nohit_addr is an address that results in a simulated
+        # NOHIT snoop response from other caches
+        self.nohit_addr = self.nohit_addresses[0]
+        # hit_addr is an address that results in a simulated
+        # HIT snoop response from other caches
+        self.hit_addr = self.hit_addresses[0]
+        # hitm_addr is an address that results in a simulated
+        # HITM snoop response from other caches
+        self.hitm_addr = self.hitm_addresses[0]
+
+    def test_exclusive(self):
+        """
+        Test reading a line in the Exclusive state.
+        Other caches return a NOHIT snoop response.
+
+        Requires the following sequence:
+        - Read line (cache miss)
+        - Read line (cache hit)
+        """
+        # Increment expected stats values based on actions we expect to
+        # modify the cache statistics object
+        # Note: these values are not determined by the actual return value
+        # of the functions being called
+        expected_reads = 0
+        expected_writes = 0
+        expected_hits = 0
+        expected_misses = 0
```

Comment:
I wanted to try out an option where these values aren't hardcoded at the end with the assertions on the stats object, but I kind of feel like the hardcoded option is more readable / understandable with regard to what we expect from the functionality of the code. So we could leave this here as an example in case we want to do this for other tests in the future, but I opted for the alternate approach in the rest of the tests.

Reply:
@nkanderson I agree, I think the hardcoded option is more readable.

Reply:
Cool, I will move forward with that approach, but leave this in for now as an example of an alternative approach. It's possible we may have more tests in the future where something like this is more appropriate.